armada_jupyter/submissions.py

```diff
@@ -122,142 +58,48 @@ def timeout_conversion(timeout: str) -> Optional[str]:
-            limits=resources.get(YMLSTR.limits),
-            requests=resources.get(YMLSTR.requests),
-        )
-
     def __repr__(self) -> str:
         return (
-            f"Submission(f{YMLSTR.name}='{self.name}', {YMLSTR.image}='{self.image}' "
-            f"{YMLSTR.armada_queue}='{self.armada_queue}', "
-            f"{YMLSTR.armada_priority}={self.armada_priority}, "
-            f"{YMLSTR.timeout}='{self.timeout}', {YMLSTR.resources}={self.resources})"
-        )
-
-    def to_podspec(self) -> core_v1.PodSpec:
-        """
-        Converts a Submission to a kubernetes PodSpec
-        """
-
-        requests = {}
-        limits = {}
-
-        if self.resources:
-            requests = {
-                YMLSTR.cpu: self.resources.requests.cpu,
-                YMLSTR.memory: self.resources.requests.memory,
-            }
-
-            limits = {
-                YMLSTR.cpu: self.resources.limits.cpu,
-                YMLSTR.memory: self.resources.limits.memory,
-            }
-
-            if self.resources.requests.nvidia_gpu:
-                requests[YMLSTR.nvidia_gpu] = self.resources.requests.nvidia_gpu
-
-            if self.resources.requests.amd_gpu:
-                requests[YMLSTR.amd_gpu] = self.resources.requests.amd_gpu
-
-            if self.resources.limits.nvidia_gpu:
-                limits[YMLSTR.nvidia_gpu] = self.resources.limits.nvidia_gpu
-
-            if self.resources.limits.amd_gpu:
-                limits[YMLSTR.amd_gpu] = self.resources.limits.amd_gpu
-
-        return core_v1.PodSpec(
-            containers=[
-                core_v1.Container(
-                    name=self.name,
-                    image=self.image,
-                    securityContext=core_v1.SecurityContext(runAsUser=1000),
-                    resources=core_v1.ResourceRequirements(
-                        requests=requests, limits=limits
-                    ),
-                )
-            ]
+            f"Submission(queue={self.queue}, job_set_id={self.job_set_id}, "
+            f"timeout={self.timeout}, jobs={self.jobs})"
         )
 
 
-def k8s_resources_from_dict(
-    limits: Optional[Dict[str, Union[str, int]]] = None,
-    requests: Optional[Dict[str, Union[str, int]]] = None,
-) -> K8sResources:
+def convert_to_submission(file: str) -> Submission:
     """
-    Creates a K8sResources object from a dict
+    Converts a yaml file into a Submission object
     """
 
-    changed: Dict[str, Optional[K8sResourceOptions]] = {}
+    with open(file, "r", encoding="utf-8") as stream:
+        data = yaml.safe_load(stream)
 
-    if limits is None or requests is None:
-        raise ValueError("Must specify both limits and requests")
+    jobs = []
 
-    # Use a loop to save repeated code.
-    for key, needs_changed in [(YMLSTR.limits, limits), (YMLSTR.requests, requests)]:
-        if needs_changed is None:
-            changed[key] = None
-            continue
+    for job in data["jobs"]:
```

Comment:
This doesn't seem like the whole Job object?  Aren't we missing queue, jobset, labels and annotations?

Reply:
Yes, that is correct, in the armada config, this is actually how it is layed out 
```yml
queue: default
jobSetId: job-set-1
timeout: "100s"
jobs:
  - priority: 1
```

Reply:
If you have a better suggestion for naming, i am all ears

Reply:
What about labels and annotations?

Reply:
I didn't include them, since they are two specific things you mentioned I will add them